incubator/command-config/src/main/java/io/aklivity/zilla/runtime/command/config/internal/openapi/http/proxy/OpenApiHttpProxyConfigGenerator.java

```diff
@@ -0,0 +1,496 @@
+        return namespace;
+    }
+
+    private HttpOptionsConfigBuilder<BindingConfigBuilder<NamespaceConfigBuilder<NamespaceConfig>>> injectHttpServerOptions(
+        HttpOptionsConfigBuilder<BindingConfigBuilder<NamespaceConfigBuilder<NamespaceConfig>>> options)
+    {
+        if (isJwtEnabled)
+        {
+            options
+                .authorization()
+                    .name("jwt0")
+                    .credentials()
+                        .header()
+                            .name("authorization")
+                            .pattern("Bearer {credentials}")
+                            .build()
+                    .build()
+                .build();
+        }
+        return options;
+    }
+
+    private BindingConfigBuilder<NamespaceConfigBuilder<NamespaceConfig>> injectHttpServerRoutes(
+        BindingConfigBuilder<NamespaceConfigBuilder<NamespaceConfig>> binding)
+    {
+        for (String path : openApi.paths.keySet())
+        {
+            PathItem2 item = PathItem2.of(openApi.paths.get(path));
+            for (String method : item.methods().keySet())
+            {
+                binding
+                    .route()
+                    .exit("http_client0")
+                    .when(HttpConditionConfig::builder)
+                        .header(":path", path.replaceAll("\\{[^}]+\\}", "*"))
+                        .header(":method", method)
+                        .build()
+                    .inject(route -> injectHttpServerRouteGuarded(route, item, method))
+                    .build();
+            }
+        }
+        return binding;
+    }
+
+    private RouteConfigBuilder<BindingConfigBuilder<NamespaceConfigBuilder<NamespaceConfig>>> injectHttpServerRouteGuarded(
+        RouteConfigBuilder<BindingConfigBuilder<NamespaceConfigBuilder<NamespaceConfig>>> route,
+        PathItem2 item,
+        String method)
+    {
+        List<Map<String, List<String>>> security = item.methods().get(method).security;
+        if (security != null)
+        {
+            for (Map<String, List<String>> securityItem : security)
+            {
+                for (String securityItemLabel : securityItem.keySet())
+                {
+                    if (isJwtEnabled && "jwt".equals(securitySchemes.get(securityItemLabel)))
+                    {
+                        route
+                            .guarded()
+                                .name("jwt0")
+                                .inject(guarded -> injectGuardedRoles(guarded, securityItem.get(securityItemLabel)))
+                                .build();
+                    }
+                }
+            }
+        }
+        return route;
+    }
+
+    private GuardedConfigBuilder<RouteConfigBuilder<BindingConfigBuilder<NamespaceConfigBuilder<NamespaceConfig>>>>
+        injectGuardedRoles(
+        GuardedConfigBuilder<RouteConfigBuilder<BindingConfigBuilder<NamespaceConfigBuilder<NamespaceConfig>>>> guarded,
+        List<String> roles)
```

Comment:
Perhaps we can simplify this signature using:
```
private <C> GuardedConfigBuilder<C> injectGuardedRoles(
    GuardedConfigBuilder<C> guarded,
    List<String> roles)
```

Reply:
done